synapse/storage/databases/main/roommember.py

```diff
@@ -848,19 +847,21 @@ async def get_joined_users_from_state(
 
         assert state_group is not None
         with Measure(self._clock, "get_joined_users_from_state"):
-            return await self._get_joined_users_from_context(
-                room_id, state_group, state, context=state_entry
-            )
+            return await self._get_joined_users_from_context(state_group, state)
 
-    @cached(num_args=2, iterable=True, max_entries=100000)
     async def _get_joined_users_from_context(
         self,
-        room_id: str,
         state_group: Union[object, int],
         current_state_ids: StateMap[str],
-        event: Optional[EventBase] = None,
-        context: Optional["_StateCacheEntry"] = None,
     ) -> Dict[str, ProfileInfo]:
+        """
+        For a given state_group, get a map of user ID to profile information for users in the room.
+
+        This method doesn't do any fetching of data itself and instead checks various
+        caches for the relevant data before passing any remaining missing event IDs to
+        `_get_joined_profiles_from_event_ids` which does the actual data fetching.
+        """
+
         # We don't use `state_group`, it's there so that we can cache based
         # on it. However, it's important that it's never None, since two current_states
```

Comment:
we no longer need `state_group` at all.

Reply:
Removed alongwith the state entry arg in: https://github.com/matrix-org/synapse/pull/13569/commits/5f99fe89ce41e1268f44a2740e57d21ff2c33b77